internal/controller/volsync/vshandler.go

```diff
@@ -1236,29 +1303,102 @@ func (v *VSHandler) DeleteRS(pvcName string, pvcNamespace string) error {
 	return nil
 }
 
-//nolint:nestif
-func (v *VSHandler) DeleteRD(pvcName string, pvcNamespace string) error {
-	// Remove a ReplicationDestination by name that is owned (by parent vrg owner)
+func (v *VSHandler) removeRDAsOwnerFromPVC(
+	rd *volsyncv1alpha1.ReplicationDestination,
+	pvcName, pvcNamespace string,
+) error {
+	// Removing the ownership should and can occur only if the resources in question reside in the same namespace.
+	// Otherwise, we exit early
+	if rd.Namespace != pvcNamespace {
+		return nil
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	key := types.NamespacedName{
+		Namespace: pvcNamespace,
+		Name:      pvcName,
+	}
+
+	if err := v.client.Get(v.ctx, key, pvc); err != nil {
+		v.log.Error(err, "Failed to get PVC for disowning", "pvc", key)
+
+		return err
+	}
+
+	updated := false
+
+	newRefs := []metav1.OwnerReference{}
+
+	for _, ref := range pvc.OwnerReferences {
+		if !(ref.Kind == "ReplicationDestination" && ref.Name == rd.Name && ref.UID == rd.UID) {
+			continue
+		}
+
+		newRefs = append(newRefs, ref)
+		pvc.OwnerReferences = newRefs
+		updated = true
+	}
+
+	if updated {
+		if err := v.client.Update(v.ctx, pvc); err != nil {
+			v.log.Error(err, "Failed to remove RD ownership from PVC", "pvc", key)
+
+			return err
+		}
+
+		v.log.Info("Removed RD ownership from PVC", "pvc", key)
+	} else {
+		v.log.Info("No RD ownership found on PVC", "pvc", key)
+	}
+
+	return nil
+}
+
+func (v *VSHandler) cleanupRD(rd *volsyncv1alpha1.ReplicationDestination, pvcName, pvcNamespace string,
+	skipPVCDisownership bool,
+) error {
+	// Step 1: Disown PVC, unless skipped
+	if !skipPVCDisownership {
+		if err := v.removeRDAsOwnerFromPVC(rd, pvcName, pvcNamespace); err != nil {
+			v.log.Error(err, "Failed to disown PVC before deleting RD", "rd", rd.GetName())
+
+			return err
+		}
+	}
+
+	// Step 2: Delete local RS if needed
+	if v.IsCopyMethodDirect() {
+		if err := v.deleteLocalRDAndRS(rd); err != nil {
+			return err
+		}
+	}
+
+	// Step 3: Delete the RD itself
+	if err := v.client.Delete(v.ctx, rd); err != nil {
+		v.log.Error(err, "Error cleaning up ReplicationDestination", "name", rd.GetName())
+
+		return err
+	}
+
+	v.log.Info("Deleted ReplicationDestination", "name", rd.GetName())
+
+	return nil
+}
+
+func (v *VSHandler) DeleteRD(pvcName, pvcNamespace string, skipPVCDisownership bool) error {
 	currentRDListByOwner, err := v.listRDByOwner(pvcNamespace)
 	if err != nil {
 		return err
 	}
 
+	expectedRDName := getReplicationDestinationName(pvcName)
```

Comment:
If we know the namespaces and name of the rd, why do we list all ads in the namespace?